Can read serial input

NNewbauer opened the pull request:
Fixes #234

**What was changed?**

We now have the ability to take data from serial inputs. I kind of strayed from the actual issue, simply because I need more information on the Data Collection page before I feel comfortable going into it. However, I have enabled the serial data to be read as a string of numbers rather than a list of encrypted data. I also added the ability to see the serial inputs while configuring devices on the setting page.

**Why was it changed?**

This was changed simply because we couldn't access the serial inputs at all until now. The big change that I made outside of decrypting the serial input was the ability to see your inputs in the setting page. I did this because if the wrong inputs are put in during the configuration stage, then they can't access their inputs, so by giving a preview of the serial input it nips this problem in the bud rather than it being an issue down the road.

**How was it changed?**

Jack and I realized that device input was a huge issue when trying to decrypt serial inputs. I temporarily hard coded caliper settings into the serial_port_connection so it can always be read during user testing (this is a super easy thing to undo). I created a new method that can read these inputs and output them as a string for other classes to use. Using this method, I created a button in the settings page that can read the input by creating an instance of that device and pulling the reading from it and printing it into the settings page. The .read_data() from the SerialPortConnection() class can easily be used for data collection down the road now.

**Screenshots that show the changes (if applicable):**

moshernoah commented on exports/Test_Experiment/animal_rfid.csv:
These files within the exports folder should not have been committed. The code (related to issue #224) has also been updated to not create these files in a manner that GitHub will see them.

moshernoah commented on settings/serial ports/.csv:
Minor changes to ensure the Caliper Serial Device works. Will need to ensure these are editable in future state for other serial devices that will be used.

moshernoah commented on shared/serial_port_controller.py:
```diff
@@ -96,10 +96,48 @@ def close_reader_port(self):
             self.reader_port.close()
 
-    def read_info(self):
+    def open_reader_port(self, port_name):
```
Properly using the saved settings from the .csv file to auto-populate serial settings. Good stuff.

moshernoah commented on shared/serial_port_controller.py:
```diff
@@ -188,6 +226,11 @@ def retrieve_setting(self, settings):
-    
+
+
+if __name__ == "__main__":
```
This prints a lot of the test print statements to ensure code functionality. Will this be needed in the future state?

moshernoah commented on shared/serial_port_settings.py:
```diff
@@ -178,7 +179,21 @@ def edit_page(self, tab: str):
+        self.test_button.grid(row=12, column=2, padx=20, pady=5, sticky="ns")
+
+        self.data_text = CTkLabel(self.edit_region, height=25, width=50, text="Waiting for Data...")
+        self.data_text.grid(row=13, column=2, columnspan=3, padx=20, pady=3, sticky="ew")
```
This code is used to do a test reading of the serial device. Does it work more than once to allow for setting changes if needed? It looks like it may only attempt to read from the device exactly once.

moshernoah reviewed (commented):
Overall, changes look good. Not everything is ready, as the test input button seems like it may have issues in the future.

kate-holdener commented on settings/serial ports/Caliper.csv:
You have a space in this directory name, that's going to be a problem on some systems. Please don't use spaces in directory or file names.

jackbelyeu replied:
I can make this a separate issue. There are years of development with this folder name, so changing it across the entire app will be more involved.

NNewbauer commented:
# Update
# Fixes #235

## What was changed?

We now have the ability to take data from serial inputs. I kind of strayed from the actual issue, simply because I need more information on the Data Collection page before I feel comfortable going into it. However, I have enabled the serial data to be read as a string of numbers rather than a list of encrypted data. I also added the ability to see the serial inputs while configuring devices on the setting page.

UPDATE: Had to change how saved device config. files are opened in order to be used for serial input correctly. Also added readings from serial input to be input in data collection.

## Why was it changed?

This was changed simply because we couldn't access the serial inputs at all until now. The big change that I made outside of decrypting the serial input was the ability to see your inputs in the setting page. I did this because if the wrong inputs are put in during the configuration stage, then they can't access their inputs, so by giving a preview of the serial input it nips this problem in the bud rather than it being an issue down the road.

UPDATE: The data from a saved serial device configuration was opened in a way that didn't match up with how methods were calling them in an array. Since the config. files were such an issue, they absolutely had to be fixed before implementing the serial devices for the data collection without hardcoding settings in (which we obviously don't want). Data collection was not functional until now, which is possibly the most important part of this program, so getting that up and running was vital.

## How was it changed?

Jack and I realized that device input was a huge issue when trying to decrypt serial inputs. I temporarily hard coded caliper settings into the serial_port_connection so it can always be read during user testing (this is a super easy thing to undo). I created a new method that can read these inputs and output them as a string for other classes to use. Using this method, I created a button in the settings page that can read the input by creating an instance of that device and pulling the reading from it and printing it into the settings page. The .read_data() from the SerialPortConnection() class can easily be used for data collection down the road now.

UPDATE: I had to do a BUNCH of rework to the serial_port_controller file in order for this to work. Data from the config. files were not being read in a way that could be called back to the device so it was essentially making the serial input devices completely useless. Had to call a method that would read through .csv config. files and correctly sort the data into a list that could be properly read by the program in order to plug it all into the serial port in order for data collection. Data collection sorted itself out fairly quickly from there, we just plugged in code that reads the first two readings of the serial device in order for automated measurements rather than having to hit a button in the software to get a measurement.

jackbelyeu commented:
Commits f763705 and 8560dad were authored by @NNewbauer, but accidentally pushed on my account.
